fix(cc): move user provided `-isystem` paths to front of command

The-Briel-Deal opened the pull request:
Problem: Currently if you pass a `-isystem` argument to `zig cc`, the include path will be searched after all of the libc, libcpp, and native system paths. However if you run the same command against `clang` and `gcc` you'll see that any user provided `-isystem` paths are at the very front.

Solution: Pull out user provided `-isystem` from `mod.cc_argv` and `comp.global_cc_argv` at the start of `addCCArgs()`, append these paths as `-isystem` args just before we append the native system paths. Finally we append the rest of the cc args in the same place.

Fixes: https://github.com/ziglang/zig/issues/24243

alexrp commented:
This doesn't seem quite correct; with this change, C flags on modules will no longer be able to override any default flags that we add in `addCCArgs()`.

andrewrk commented:
That's right - those are at the end on purpose so they can override everything else.

The proper fix is to adjust the parsing in the CLI to account for isystem args without piggybacking on the general last resort overriding flag set.

The-Briel-Deal committed:
fix(cc): move user provided `-isystem` paths to front of command

Problem: Currently if you pass a `-isystem` argument to `zig cc`, the
include path will be searched after all of the libc, libcpp, and native
system paths. However if you run the same command against `clang` and
`gcc` you'll see that any user provided `-isystem` paths are at the very
front.

Solution: Push all user provided arguments to the arg list at the start
of `addCCArgs()`. This makes it so that any compiler arguments that rely
on the order of arguments passed will prioritize the user provided
values first.

Fixes: https://github.com/ziglang/zig/issues/24243

The-Briel-Deal commented:
Hey @alexrp! I just wanted to follow up on this, this fix should be ready for review if you have a minute some time in the next few days.

andrewrk commented:
This pull request is not ready for review because:
* It has conflicts that must be resolved via rebasing against latest origin/master.
* It is not passing the CI tests.

Closing since there has been no activity for 30+ days.
